src/biogeophys/QSatMod.F90

```diff
@@ -77,91 +75,56 @@ subroutine QSat (T, p, es, esdT, qs, qsdT)
     implicit none
     real(r8), intent(in)  :: T        ! temperature (K)
     real(r8), intent(in)  :: p        ! surface atmospheric pressure (pa)
-    real(r8), intent(out) :: es       ! vapor pressure (pa)
-    real(r8), intent(out) :: esdT     ! d(es)/d(T)
     real(r8), intent(out) :: qs       ! humidity (kg/kg)
-    real(r8), intent(out) :: qsdT     ! d(qs)/d(T)
+    real(r8), intent(out), optional :: es       ! vapor pressure (pa)
+    real(r8), intent(out), optional :: qsdT     ! d(qs)/d(T)
+    real(r8), intent(out), optional :: esdT     ! d(es)/d(T)
     !
     ! !LOCAL VARIABLES:
-    real(r8) :: T_limit
+    real(r8) :: es_local    ! local version of es (in case es is not present)
+    real(r8) :: esdT_local  ! local version of esdT (in case esdT is not present)
     real(r8) :: td,vp,vp1,vp2
     !-----------------------------------------------------------------------
 
-    T_limit = T - SHR_CONST_TKFRZ
-    if (T_limit > 100.0_r8) T_limit=100.0_r8
-    if (T_limit < -75.0_r8) T_limit=-75.0_r8
+    td = T - SHR_CONST_TKFRZ
+    if (td > 100.0_r8) td=100.0_r8
+    if (td < -75.0_r8) td=-75.0_r8
```

Comment:
Would this be an answer-changing modification?
```suggestion
    td = min(100.0_r8, max(-75.0_r8, T - SHR_CONST_TKFRZ))
```

Reply:
Done. I personally have a really hard time with mins & maxes so I don't tend to jump to them as a solution, but I do see how this more clearly shows the intent of this code - at least to someone who has less of a hard time with mins and maxes than I do :-)

Reply:
By the way, I *think* this should be a bit-for-bit change, but I'll back it out if baseline comparisons fail.